inmem_endpoint.go

```diff
@@ -0,0 +1,100 @@
+package metrics
+
+import (
+	"fmt"
+	"net/http"
+	"sort"
+)
+
+// MetricsSummary holds a roll-up of metrics info for a given interval
+type MetricsSummary struct {
+	Timestamp string
+	Gauges    []GaugeValue
+	Points    []PointValue
+	Counters  []SampledValue
+	Samples   []SampledValue
+}
+
+type GaugeValue struct {
+	Name   string
+	Hash   string `json:"-"`
+	Value  float32
+	Labels []Label
+}
+
+type PointValue struct {
+	Name   string
+	Points []float32
+}
+
+type SampledValue struct {
+	Name string
+	Hash string `json:"-"`
+	*AggregateSample
+	Mean   float64
+	Stddev float64
+	Labels []Label
+}
+
+// DisplayMetrics returns a summary of the metrics from the most recent finished interval.
+func (i *InmemSink) DisplayMetrics(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	data := i.Data()
+
+	var interval *IntervalMetrics
+	n := len(data)
+	switch {
+	case n == 0:
+		return nil, fmt.Errorf("no metric intervals have been initialized yet")
+	case n == 1:
+		interval = i.intervals[0]
+	default:
+		interval = i.intervals[n-2]
+	}
+
+	summary := MetricsSummary{
+		Timestamp: interval.Interval.String(),
```

Comment:
In Consul we ended up putting all of the timestamps we display into UTC so people don't have to mess with time zones - this might be good to do:

https://github.com/hashicorp/consul/blob/v0.9.0/agent/consul/acl_replication.go#L255-L258

I'm not sure if we want to shred any of the precision to get them to display uniformly (or if that matters here).